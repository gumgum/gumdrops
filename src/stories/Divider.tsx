import React from 'react';
import cx from 'classnames';

export interface DividerProps extends React.HTMLAttributes<Element> {
    label?: string;
    centered?: boolean;
    collapsible?: boolean;
    open?: boolean;
}

const baseClass = 'gds-divider';

export const Divider: React.FC<DividerProps> = ({
    label,
    centered,
    collapsible,
    open,
    className,
    ...otherProps
}) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${centered}`]: centered,
        [`${baseClass}--${collapsible} -cursor--pointer`]: collapsible
    });

    const arrowClasses = cx(`${baseClass}__arrow`, {
        [`${baseClass}__arrow--collapse`]: !open
    });

    return (
        <div className={rootClass} {...otherProps}>
            {centered && <span className="gds-divider__line" />}
            <span className="gds-divider__label gds-form-group__label">{label}</span>
            <span className="gds-divider__line" />
            {collapsible && <span className={arrowClasses} />}
        </div>
    );
};
