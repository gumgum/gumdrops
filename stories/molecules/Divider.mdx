import { Story, Source, Props } from '@storybook/addon-docs/blocks';
import Divider from '../../components/molecules/Divider';

# Divider

The standard `<Divider>` component is used to separate distinct segments of content.

If the divider is collapsible, pass in a callback function that sets the state, and include the state in your component.

## Full Example

```jsx
const [isOpen, setIsOpen] = useState(false);
return (
    <div>
        <Divider
            label="Collapsible Label"
            collapsible
            open={isOpen}
            callback={() => setIsOpen(!isOpen)}
        />
        {isOpen && <div>Content to show when toggled open</div>}
    </div>
);
```

## Component Example

<Story id="molecules-divider--default" />

## Source

<Source id="molecules-divider--default" />

## Props

<Props of={Divider} />
