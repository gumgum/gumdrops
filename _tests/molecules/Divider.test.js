import renderer from 'react-test-renderer';
import React from 'react';
import Divider from '../../components/molecules/Divider';

const defaultProps = {
    label: 'Divider Label',
    centered: false,
    collapsible: false,
    open: false,
    callback: jest.fn(),
    style: { width: 150 }
};

test('Expect <Divider> to render properly', () => {
    const tree = renderer.create(<Divider {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <Divider> to render open', () => {
    const props = {
        ...defaultProps,
        collapsible: false,
        open: true
    };
    const tree = renderer.create(<Divider {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
