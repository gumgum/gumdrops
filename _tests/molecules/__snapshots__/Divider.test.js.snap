// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Divider> to render open 1`] = `
<div
  className="gds-divider"
  onClick={[MockFunction]}
  style={
    {
      "width": 150,
    }
  }
>
  <span
    className="gds-divider__label gds-form-group__label"
  >
    Divider Label
  </span>
  <span
    className="gds-divider__line"
  />
</div>
`;

exports[`Expect <Divider> to render properly 1`] = `
<div
  className="gds-divider"
  onClick={[MockFunction]}
  style={
    {
      "width": 150,
    }
  }
>
  <span
    className="gds-divider__label gds-form-group__label"
  >
    Divider Label
  </span>
  <span
    className="gds-divider__line"
  />
</div>
`;
